weather_data.php

```diff
@@ -3,6 +3,36 @@
 global $map, $fork;
 header('Content-Type: application/json');
 // init map
+$swLat = !empty($_POST['swLat']) ? $_POST['swLat'] : 0;
+$neLng = !empty($_POST['neLng']) ? $_POST['neLng'] : 0;
+$swLng = !empty($_POST['swLng']) ? $_POST['swLng'] : 0;
+$neLat = !empty($_POST['neLat']) ? $_POST['neLat'] : 0;
+$oSwLat = !empty($_POST['oSwLat']) ? $_POST['oSwLat'] : 0;
+$oSwLng = !empty($_POST['oSwLng']) ? $_POST['oSwLng'] : 0;
+$oNeLat = !empty($_POST['oNeLat']) ? $_POST['oNeLat'] : 0;
+$oNeLng = !empty($_POST['oNeLng']) ? $_POST['oNeLng'] : 0;
+
+$useragent = isset($_SERVER['HTTP_USER_AGENT']) ? $_SERVER['HTTP_USER_AGENT'] : '';
+if (empty($swLat) || empty($swLng) || empty($neLat) || empty($neLng) || preg_match("/curl|libcurl/", $useragent)) {
+    http_response_code(400);
+        die();
+}
+if ($maxLatLng > 0 && ((($neLat - $swLat) > $maxLatLng) || (($neLng - $swLng) > $maxLatLng))) {
+    http_response_code(400);
+        die();
+}
+if (!validateToken($_POST['token'])) {
+    http_response_code(400);
+        die();
+}
+if ((! $noDiscordLogin || ! $noNativeLogin) && !empty($_SESSION['user']->id)) {
+    $info = $manualdb->query("SELECT expire_timestamp FROM users WHERE id = :id", [":id" => $_SESSION['user']->id])->fetch();
+        if ($info['expire_timestamp'] !== $_SESSION['user']->expire_timestamp) {
```

Comment:
select session_id instead of expire_timestamp and compare `$info['session_id'] !== $_COOKIE["LoginCookie"]` so it doesnt error out when using $manualAccessLevel